client/allocrunner/taskrunner/template/template_test.go

```diff
@@ -806,6 +806,69 @@ OUTER:
 	}
 }
 
+// Tests an edge case where a task has multiple templates and the client is restarted.
+// In this case, vault may re-render and overwrite some fields in the first render event
```

Comment:
Vault? I think we're saying that the upstream service (for example Vault) might change and force a render, but it reads a little weird.

Reply:
Whoops I mean't consule-template.  Good catch!